src/js/contexts/AnalyticsContext/AnalyticsContext.js

```diff
@@ -0,0 +1,40 @@
+import React, { useCallback, useContext, useEffect, useRef } from 'react';
+
+export const AnalyticsContext = React.createContext(() => {});
+
+export const useAnalytics = () => useContext(AnalyticsContext);
+
+export const AnalyticsProvider = ({ onAnalytics, children }) => {
+  const pathInfoRef = useRef({
+    lastUrl: window.location.href,
+    observer: undefined,
+  });
+
+  const sendAnalytics = useCallback(
+    (data) => onAnalytics && onAnalytics(data),
+    [onAnalytics],
+  );
+
+  useEffect(() => {
+    let observer;
+    if (onAnalytics) {
+      observer = new window.MutationObserver(() => {
+        const url = window.location.href;
+        if (url !== pathInfoRef.current.lastUrl) {
+          pathInfoRef.current.lastUrl = url;
+          sendAnalytics({ type: 'pageView', data: url });
+        }
+      });
+      observer.observe(document, { subtree: true, childList: true });
+    }
+    pathInfoRef.current.observer = observer;
```

Comment:
Why do we need to save the `observer` in the `pathInfoRef`?

Reply:
Probably don't. I originally put it there so consumers would have access to it, but I don't think it's useful